DOCS-2881: Revamp Micro-RDK docs (#4241)
viamrobotics/docs: docs/operate/reference/viam-micro-server/_index.md

# SMEs: Nicolas M., Gautham V., Andrew M.
---

> [The Micro-RDK](https://github.com/viamrobotics/micro-rdk/) is the lightweight version of Viam's full [Robot Development Kit (RDK)](https://github.com/viamrobotics/rdk/tree/main).
> It is written in Rust and designed for resource-limited embedded systems (ESP32 microcontrollers) that cannot run the fully-featured [`viam-server`](/operate/reference/viam-server/).

Author: I'll own that this is partially my OG language (collaboratively)--not new in this PR but I did put it there before haha.
- I agree that especially before you read the second sentence, it's not explicitly clear that the Micro-RDK is only for ESP32.
- I had included Rust because we don't use Rust for anything else in Viam, and micro modules must be written in Rust. But I guess it doesn't really need to be there and taking it out allows everything to fit into one clearer sentence
- "Lightweight" and "full" are there to emphasize that the Micro-RDK doesn't have nearly all the capabilities of the RDK, but I like your "fully-featured" wording--that makes it more clear than "full"

[nit] FYI we don't use e.g. (or i.e.) but rather "for example" or similar. In any case, micro actually *only* runs on ESP32 at this time, AFAIK, so it's not just an example but rather the only thing.

```suggestion
[The Micro-RDK](https://github.com/viamrobotics/micro-rdk/) is the lightweight version of Viam's [Robot Development Kit (RDK)](https://github.com/viamrobotics/rdk/tree/main) designed for embedded systems (ESP32 microcontrollers) that do not have the resources to run the fully-featured [`viam-server`](/operate/reference/viam-server/).
```

Microcontrollers do not have full operating systems like single-board computers and general-purpose computers.

> This means that microcontrollers can only run one "program" at a time, and a microcontroller must be flashed with firmware containing the entire logic to run.

Author: This is a great suggestion ty

> `viam-micro-server` is the pre-built firmware built from the Micro-RDK and a [default set](/operate/get-started/setup-micro/#about-esp32-microcontroller-setup) of {{< glossary_tooltip term_id="module" text="modules" >}}, provided for a convenient quick-start.

Author: I think "fast quick-start" is redundant though.
I didn't want to say "provided for your convenience" because I don't like that phrase, but was essentially trying to get across why we even bothered to make `viam-micro-server`. Might leave as-is unless you have another suggestion on hand!

> `viam-micro-server` is the pre-built firmware built from the Micro-RDK and a [default set](/operate/get-started/setup-micro/#about-esp32-microcontroller-setup) of {{< glossary_tooltip term_id="module" text="modules" >}}, provided for a convenient quick-start.

Reviewer: Updated the suggestion. It removes the awkward 'fast quick' wording, but kills the 'quick-start' hyphenation, which... is fine by me. But if you don't like it you can always leave it as-is!

> `viam-micro-server` is the pre-built firmware built from the Micro-RDK and a [default set](/operate/get-started/setup-micro/#about-esp32-microcontroller-setup) of {{< glossary_tooltip term_id="module" text="modules" >}}, provided for a convenient quick-start.

Author: Okay sure I'll apply this one!
